ouroboros-network/src/Ouroboros/Network/Mux/Types.hs

```diff
@@ -181,14 +229,27 @@ data PerMuxSharedState ptcl m = PerMuxSS {
     dispatchTable :: MiniProtocolDispatch ptcl m
   -- | Egress queue, shared by all miniprotocols
   , tsrQueue      :: TBQueue m (TranslocationServiceRequest ptcl m)
-  -- | Timestamp and send MuxSDU
-  , write         :: MuxSDU ptcl -> m (Time m)
-  -- | Read a MuxSDU
-  , read          :: m (MuxSDU ptcl, Time m)
-  -- | Return a suitable MuxSDU payload size
-  , sduSize       :: m Word16
+  , bearer        :: MuxBearer ptcl m
   }
 
+data MuxBearerState = Larval
+                    | Connected
+                    | Mature
+                    | Dying
+                    | Dead
+                    deriving (Eq, Show)
+
+data MuxBearer ptcl m = MuxBearer {
```

Comment:
Maybe it would be good to move `Socket` and `Pipe` to `Ouroboros.Network.Mux.Bearer.{Socket,Pipe,Queue}` (the `Queue` which is in tests right now (it could be only included in tests), and we can put the `MuxBearer` related functions in `Mux.Bearer`.  But it's probably better to do that in another PR.

Reply:
I'm not sure this is a good idea. Pipe will get its own version of connection handling to support multiple Mux bearers over the same pair of pipes. If it turns out to be a lot of code I'd like to keep Pipe and Socket separate.

Reply:
I have realized that I misunderstood your comment.
Moving Socket, Pipe and Queue into Mux.Bearer is a good idea.